serverv2/cometbft/server.go

```diff
@@ -0,0 +1,119 @@
+package cometbft
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/log"
+	abciserver "github.com/cometbft/cometbft/abci/server"
+	abci "github.com/cometbft/cometbft/abci/types"
+	cmtcfg "github.com/cometbft/cometbft/config"
+	cometservice "github.com/cometbft/cometbft/libs/service"
+	"github.com/cometbft/cometbft/node"
+	"github.com/cometbft/cometbft/p2p"
+	pvm "github.com/cometbft/cometbft/privval"
+	"github.com/cometbft/cometbft/proxy"
+	cmttypes "github.com/cometbft/cometbft/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	cometlog "github.com/cosmos/cosmos-sdk/serverv2/cometbft/log"
+	"github.com/cosmos/cosmos-sdk/serverv2/cometbft/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+)
+
+type HasCometBFTServer interface {
+	types.ProtoApp
+
+	// RegisterTendermintService registers the gRPC Query service for CometBFT queries.
+	RegisterTendermintService(client.Context)
+}
+
+type Config struct {
+	Transport  string
+	Addr       string
+	Standalone bool
+
+	CmtConfig *cmtcfg.Config
+}
+
+type CometBFTServer struct {
+	Node   *node.Node
+	app    abci.Application
+	logger log.Logger
+
+	config    Config
+	service   cometservice.Service
+	cleanupFn func()
+}
+
+func NewCometBFTServer(clientCtx client.Context, logger log.Logger, app HasCometBFTServer, cfg Config) *CometBFTServer {
```

Comment:
```suggestion
func NewCometBFTServer( logger log.Logger, app HasCometBFTServer, cfg Config) *CometBFTServer {
```

we should remove client context and have it only take what is needed

appamanger, mempool, prepare/process/vote extension handlers, and some auxillary stuff. A lof of the other stuff hasnt been written yet so its a bit harder to expect it. We can start with simple interfaces and expand on them as we need